src/storage/buffer_manager/buffer_managed_file_handle.cpp

```diff
@@ -150,7 +152,8 @@ void BufferManagedFileHandle::clearWALPageVersionIfNecessary(common::page_idx_t
     }
     createPageVersionGroupIfNecessary(pageIdx);
     setWALPageVersionNoLock(pageIdx, UINT32_MAX);
-    releasePageLock(pageIdx);
+    // TODO(GUODONG): Why do we release lock here? Where was it locked?
```

Comment:
Added the lock acquiring.

Reply:
Actually I kept this function as it is (didn't add the lock acquiring), and kept my comment here. There seems to be a deadlock problem after I acquired page lock before `setWALPageVersionNoLock(pageIdx, UINT32_MAX);`, which I hadn't look into yet. `SetNodeStructuredPropTransactionTest.SetVeryLongStringErrorsTest` can reproduce the problem.

Reply:
Let's not make any changes to this function without understanding this. How do we know for example that the releasePageLock is not necessary? If you can't figure this out or it looks like it will take a long time for you to figure it out, just open an issue and assign to me but also don't remove the call to releasePageLock. In general it's not a good idea to squeeze these changes that are not related to your PR into the PR anyways.